DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepositoryND.cs

```diff
@@ -233,6 +252,29 @@ public DeviceRegistryRepositoryND(IConfigurationProvider configProvider, IDocDbR
             };
         }
 
+        public async Task<DeviceListQueryResultND> GetDeviceListND(DeviceListQuery query)
+        {
+            List<dynamic> deviceList = await GetAllDevicesAsync();
+
```

Comment:
Thing we can convert the deviceList in GetAllDevicesAsync to DeviceND objects and return that list